test/WebSites/RazorWebSite/Services/TestHeadTagHelperComponent.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+
+namespace RazorWebSite
+{
+    public class TestHeadTagHelperComponent : TagHelperComponent
+    {
+        public TestHeadTagHelperComponent()
+        {
+        }
+
+        public override int Order => 1;
+
+        public override Task ProcessAsync(TagHelperContext context, TagHelperOutput output)
+        {
+            if (context.TagName == "head")
+            {
+                var customAttribute = new TagHelperAttribute("inject");
+                context.AllAttributes.TryGetAttribute("inject", out customAttribute);
+                if (customAttribute?.Value.ToString() == "true")
```

Comment:
@Eilon / @DamianEdwards it really sucks that @jbagga was forced down this path. We need a method somewhere to make stringifying a `TagHelperAttribute` correctly easier: https://github.com/aspnet/Mvc/issues/6030